### ISSUE TITLE
How do we actually use the msdn.4.5.2 rpk files?

### ISSUE THREAD

**Issue by LordZoltan** [dotnet/docfx#749]
Documentation as to how to use the rpk files from the msdn.4.5.2 package to enable cross-linking from our site to the MSDN website is pretty scarce.

http://dotnet.github.io/docfx/tutorial/links_and_cross_references.html talks about using an 'xrefMaps' property for the build section; and then mentions the RPK package, but doesn't explain exactly how you're supposed to link them.

Section 3.2 of the 'user manual' http://dotnet.github.io/docfx/tutorial/docfx.exe_user_manual.html#3-docfxjson-format lists the now obsolete 'externalReferences' build property, without descibing what the alternative is.  And, just below that, it then talks about another property, 'xref' which appears to do the same thing as the 'xrefMaps' property described in the links/cross-references section.

--

Are we supposed to be able to link RPKs directly?  Or (since I have now discovered they are actually just Zip files containing all the YMLs for the different types) do we have to extract the individual files for each of the types we wish to link?

I was hoping we'd be able to do something like:

    xrefMaps: [
      { 'files': '*.rpk' }, //referencing all RPKs in the root of the project (as installed by the package)
      { 'files': 'someOtherXrefMap.yml' }, //individual ymls (or rpks)
      { 'files': ['companyXrefMaps/*.yml'], 'exclude': 'toexclude.yml' } //exclusions
    ]

Basically - I'd like a one-stop property for configuring cross-references, supporting both yml files and zipped packages containing ymls; all specified using the standard file pattern.

In the meantime, I'd just like to know how I can easily get docfx to pick up all the `<see cref="">` links I have in my API documentation which reference .Net types because, at the moment, the words just disappear from the output, which is not good.  If it can't resolve a `<see cref` then it should probably just grab the text out of the `cref` attribute and print it as-is, rather than excluding it from the output.

Many thanks

**Comment by vwxyzh** [comment]
RPK files is obsoleted. Now we use `xref` for yaml files or zip files and it support http/https or local files.
Package `msdn.4.5.2` is also updated to zip file format now.
e.g. [our document docfx config](https://github.com/dotnet/docfx/blob/dev/Documentation/docfx.json#L36-L38)

**Comment by LordZoltan** [comment]
Thanks for your reply :)

So, I understand that the msdn.4.5.2 package has been updated in source to include the single zip file now, instead of all the RPK files - that's great - except it doesn't appear to have been published on nuget.org.  The latest version of that package still has all the rpk files.  Should it have been published?  Or is it available on a different nuget feed?

In the meantime I shall pull the source into my local fork of the docfx code, and reference it from there, similar to how you've done in the example.

Many thanks

**Comment by vwxyzh** [comment]
You can find it in [nuget](https://www.nuget.org/packages/msdn.4.5.2/0.1.0-alpha-1610081127) now.

**Comment by LordZoltan** [comment]
Brilliant - thanks!

**Comment by LordZoltan** [comment]
I've noticed that types in System.Linq.Expressions don't seem to auto link.  I'll open a new issue with examples at some point :)
